File: 3136.valid-word/solution.go
var vocals = "aeiouAEIOU"

func isValid(word string) bool {
    if len(word) < 3 {
        return false
    }
    var v, c bool
    for _, ch := range word {
        if unicode.In(ch, unicode.Digit) {
            continue
        }
        if !unicode.In(ch, unicode.Letter) {
            return false
        }
        if strings.ContainsRune(vocals, ch) {
            v = true
        } else {
            c = true
        }
    }
    return v && c
}

File: 3136.valid-word/solution.kt
class Solution {
    fun isValid(word: String): Boolean {
        return word.length >= 3 &&
            word.all { c -> c.isLetterOrDigit() } &&
            word.any { c -> c.isVowel() } &&
            word.any { c -> c.isLetter() && !c.isVowel() }
    }
}

fun Char.isVowel() = "aeiouAEIOU".indexOf(this) >= 0